scripts/e2e/cmd/runner/helper.go

```diff
@@ -93,29 +94,77 @@ func applyYaml(clientset *kubernetes.Clientset, namespaceName string, fileName s
 
 	for _, objs := range fileObjects {
 		if secret, ok := objs.(*v1.Secret); ok {
-			if _, err := clientset.CoreV1().Secrets(namespaceName).Create(secret); err != nil {
-				return err
+			nm := secret.Namespace
+			if len(nm) == 0 && len(namespaceName) != 0 {
```

Comment:
```golang
if len(nm) != 0 {
    ...
} else if len(namespaceName) != 0 {
    ...
} else {
    return errors.New("namespace is not defined for secrets")
}
```

Reply:
> ```go
> if len(nm) != 0 {
>     ...
> } else if len(namespaceName) != 0 {
>     ...
> } else {
>     return errors.New("namespace is not defined for secrets")
> }
> ```

In order to use namespace defined in yaml, yaml spec should have the namespace defined and the provided namespace should be empty.